rasa/core/featurizers/single_state_featurizer.py

```diff
@@ -40,127 +102,227 @@ class SingleStateFeaturizer:
     featurized into a list of `rasa.utils.features.Features`.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
         """Initialize the single state featurizer."""
-        # rasa core can be trained separately, therefore interpreter during training
-        # will be `RegexInterpreter`. If the model is combined with a rasa nlu model
-        # during prediction the interpreter might be different.
-        # If that is the case, we need to make sure to "reset" the interpreter.
-        self._use_regex_interpreter = False
-        self._default_feature_states = {}
-        self.action_texts = []
-        self.entity_tag_specs = []
-
-    def _create_entity_tag_specs(
-        self, bilou_tagging: bool = False
-    ) -> List[EntityTagSpec]:
-        """Returns the tag to index mapping for entities.
+        self.prepared = False
```

Comment:
Somehow, we need to get rid of the need to "prepare". The vocabulary build up should be done during initialization.

Reply:
Hm, just realised, if we move that to the init only, then when we reload this component (after persisting it) it won't be able to magically adjust to an updated domain.